### PR TITLE
Log using context to print more information

### PR DESCRIPTION
**Pull request by sunby**
We need to relate log info with module name and requestID. So I save
the logger with these fields in context. And we can pass the context
through the call stack.

Signed-off-by: sunby <bingyi.sun@zilliz.com>

/kind improvement

### CODE REVIEW

**Comment by sunby** [comment]
I test on my own machine. It will print log like 
image

File: internal/proxy/proxy.go [hunk 1]
```diff
@@ -89,34 +89,37 @@ type Proxy struct {
 	// Add callback functions at different stages
 	startCallbacks []func()
 	closeCallbacks []func()
+
+	logCtx context.Context
```
Comment (congqixia) [inline internal/proxy/proxy.go:93]: It feels weird to storage a context with logger here. Maybe a logger will be enough?

Comment (sunby) [reply internal/proxy/proxy.go:93]: In this case yes, but if we want to pass a context to another function, we should know the log context. If I dont save the context, I will create this context every time I log infos.

Comment (sunby) [reply internal/proxy/proxy.go:93]: It wil be great if there's a better practice.

Comment (congqixia) [reply internal/proxy/proxy.go:93]: `Proxy` already has a `ctx` field stored. (Not a good practice, BTW). Which context member shall we pass when invoke a utility function? My suggestion is: 
1. If proxy function needs a logger to pass on, with a context parameter, say `CreateCollection`, use context.WithValue to add the logger into context to pass on.
2. it other case, say `Init`, maybe we could either use ```context.WithValue(p.ctx, logger)```, or just use a Background one.

Comment (sunby) [reply internal/proxy/proxy.go:93]: `logCtx` is only used for saving a logger of a module. If a new grpc request arrives, we should get a ctx with logger by `WithValue`. Maybe grpc log context have other fields like requestID. I agree that storing a context is not a good practice, but if we store a logger in proxy struct and we should create a context with this logger every time we call other functions and this is a little dirty.

File: internal/proxy/task.go [hunk 2]
```diff
@@ -1183,7 +1184,15 @@ func (cct *createCollectionTask) PreExecute(ctx context.Context) error {
 
 func (cct *createCollectionTask) Execute(ctx context.Context) error {
 	var err error
+	logCtx := logutil.WithReqID(cct.ctx, cct.GetBase().GetMsgID())
+
+	logutil.Logger(logCtx).Debug("send create collection request to RootCoord")
 	cct.result, err = cct.rootCoord.CreateCollection(ctx, cct.CreateCollectionRequest)
+	logutil.Logger(logCtx).Debug("receive create collection response from RootCoord",
```
Comment (congqixia) [inline internal/proxy/task.go:1191]: Which context shall we use in line 1190, which invokes `CreateCollection` of rootcoord?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**